Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/Batching/Batching.xaml.cs

```diff
@@ -0,0 +1,240 @@
+        {
+            float totalEvalDurations = 0;
+            for (int i = 0; i < 100; i++)
+            {
+                if (navigatingAwayFromPage)
+                    break;
+                UpdateEvalProgressUI(i);
+                float evalDuration = await System.Threading.Tasks.Task.Run(() => EvaluateBatched(BatchingSession_, inputImages, batchSize));
+                totalEvalDurations += evalDuration;
+            }
+            avgBatchDuration = totalEvalDurations / 100;
+        }
+
+        private static float EvaluateBatched(LearningModelSession session, List<VideoFrame> input, int batchSize)
+        {
+            // Evaluate
+            int numBatches = input.Count / batchSize;
+            string inputName = session.Model.InputFeatures[0].Name;
+            float totalDuration = 0;
+            for (int i = 0; i < numBatches; i++)
+            {
+                if (navigatingAwayFromPage)
+                    break;
+                var binding = new LearningModelBinding(session);
+                List<VideoFrame> batch = input.GetRange(batchSize * i, batchSize);
+                var start = HighResolutionClock.UtcNow();
+                binding.Bind(inputName, batch);
+                session.Evaluate(binding, "");
+                var stop = HighResolutionClock.UtcNow();
+                var duration = HighResolutionClock.DurationInMs(start, stop);
+                totalDuration += duration;
+            }
+            return totalDuration;
+        }
+
+        private int GetBatchSizeFromBatchSizeComboBox()
+        {
+            int batchSize;
+            if (BatchSizeComboBox.SelectedIndex == 0)
+                batchSize = 5;
+            else if (BatchSizeComboBox.SelectedIndex == 1)
+                batchSize = 10;
+            else if (BatchSizeComboBox.SelectedIndex == 2)
+                batchSize = 25;
+            else
+                batchSize = 50;
+            return batchSize;
+        }
+
+        private void UpdateEvalProgressUI(int attemptNumber)
+        {
+            EvalProgressText.Text = "Attempt " + attemptNumber.ToString() + "/100";
+            EvalProgressBar.Value = attemptNumber + 1;
+        }
+
+        private void GenerateEvalResultAndUI()
+        {
+            float ratio = (1 - (avgBatchDuration / avgNonBatchedDuration)) * 100;
+            var evalResult = new EvalResult
+            {
+                nonBatchedAvgTime = avgNonBatchedDuration.ToString("0.00"),
+                batchedAvgTime = avgBatchDuration.ToString("0.00"),
+                timeRatio = ratio.ToString("0.0")
+            };
+            List<EvalResult> results = new List<EvalResult>();
+            results.Insert(0, evalResult);
+            LoadingContainer.Visibility = Visibility.Collapsed;
+            EvalResults.Visibility = Visibility.Visible;
+            StartInferenceBtn.IsEnabled = true;
+            EvalResults.ItemsSource = results;
+        }
+
+        public void StopAllEvents()
```

Comment:
>StopAllEvents [](http://example.com/codeflow?start=20&length=13)

Is there a need for this?

Cant you implement Page.OnNavigatingFrom?

Reply:
Since the page is being loaded as a frame into SampleBasePage OnNavigatingFrom only gets triggered on SampleBaseBage. From there I call this Page.StopAllEvents function